bin/asl-say

```diff
@@ -145,6 +164,29 @@ if [ -z "${NODE}" ] || [ ${NODE} -lt 1 ]; then
 	exit
 fi
 
+if [ -n "${ALT_DATE}${ALT_TIME}" ]; then
+	if [ -n "${ALT_DATE}" ]; then
+		re=^[0-9]{8}$
+		if ! [[ ${ALT_DATE} =~ $re ]]; then
+			echo "Alternate date must be YYYYMMDD"
+			exit 1
+		fi
+	else
+		ALT_DATE=$(date +%Y%m%d)
+	fi
+	if [ -n "${ALT_TIME}" ]; then
+		re=^[0-9]{4}$
+		if ! [[ ${ALT_TIME} =~ $re ]]; then
+			echo "Alternate time must be HHMM"
+			exit 1
+		fi
+	else
+		ALT_TIME=$(date +%H%M)
+	fi
+	touch -t "${ALT_DATE}${ALT_TIME}" /tmp/asl-say-date-reference
+	ALT_DATE_TIME="-r /tmp/asl-say-date-reference"
```

Comment:
I guess using a file timestamp to generate the datetime is ok, but I'm not sure it's necessary when the environment variable could be directly used in the `date` command. 
And sort of an aside because it it so unlikely to apply here, but I have seen the timestamp on a network mounted filesystem follow the system time on the fileserver which had drifted from the local system time on the machine that was making the file.

Reply:
Ok I think I see why the file is being used - it's because there are multiple invocations of `date` throughout the script. On other way to solve this without relying on the filesystem would be to store the timestamp, for instance as seconds past epoch:
`MY_TIME=$(date +%s)`
Then on the other invocations use
`date -d @$MY_TIME`

Reply:
The macOS `date` command does not support "-d".  But, it does support "-r file" and the `touch` command gave me a way to set the date/time stamp on the file so I put those pieces together.  Using "-d" will work out well.